### PR TITLE
print() is a function in Python 3

### PR DESCRIPTION
**Pull request by cclauss**
These are some "safe" changes but there will be other issues that need to be resolved including:

$ __python3 -m flake8 . --count --select=E9,F821,F822,F823 --show-source --statistics__
```
./tree/max_path_sum.py:13:19: F823 local variable 'maximum' (defined in enclosing scope on line 2) referenced before assignment
    maximum = max(maximum, left+right+root.val)
                  ^
./tree/longest_consecutive.py:43:23: F823 local variable 'maxlen' (defined in enclosing scope on line 26) referenced before assignment
    maxlen = max(cur, maxlen)
                      ^
./tree/bst/serialize_deserialize.py:21:16: F821 undefined name 'TreeNode'
        node = TreeNode(int(val))
               ^
./tree/bst/array2bst.py:16:12: F821 undefined name 'Node'
    node = Node(nums[mid])
           ^
./linkedlist/copy_random_pointer.py:17:18: F821 undefined name 'RandomListNode'
        dic[m] = RandomListNode(m.label)
                 ^
./linkedlist/copy_random_pointer.py:31:44: F821 undefined name 'RandomListNode'
    copy = collections.defaultdict(lambda: RandomListNode(0))
                                           ^
./heap/merge_sorted_k_lists.py:14:20: F821 undefined name 'ListNode'
    dummy = node = ListNode(0)
                   ^
./heap/merge_sorted_k_lists.py:31:13: F821 undefined name 'ListNode'
    dummy = ListNode(None)
            ^
./queues/queue.py:29:18: F821 undefined name 'AbstractStack'
class ArrayQueue(AbstractStack):
                 ^
./queues/queue.py:36:9: F821 undefined name 'AbstractStack'
        AbstractStack.__init__(self)
        ^
./queues/queue.py:78:23: F821 undefined name 'AbstractStack'
class LinkedListQueue(AbstractStack):
                      ^
./queues/queue.py:86:16: F821 undefined name 'front'
        if not front:
               ^
./queues/queue.py:112:25: F821 undefined name 'AbstractStack'
class HeapPriorityQueue(AbstractStack):
                        ^
./stack/longest_abs_path.py:64:18: F821 undefined name 'lengthLongestPath'
print("answer:", lengthLongestPath(st2))
                 ^
12    F821 undefined name 'ListNode'
2     F823 local variable 'maxlen' (defined in enclosing scope on line 26) referenced before assignment
14
```

### CODE REVIEW

File: googlecodejam/Fashion Show/fashion_show.py [hunk 1]
```diff
@@ -57,10 +57,10 @@ def merge_solutions(rook_solution, bishop_solution):
 
 
 def solve(board):
-    rook_board = map_board(lambda piece: 'x' if (piece == 'x' or piece == 'o') else '.', board)
-    bishop_board = map_board(lambda piece: '+' if (piece == '+' or piece == 'o') else '.', board)
+    rook_board = map_board(lambda piece: 'x' if piece in 'xo' else '.', board)
+    bishop_board = map_board(lambda piece: '+' if piece in '+o' else '.', board)
     rook_solution = solve_rooks(rook_board)
-    bishop_solution = solve_bishops(bishop_board)
+    bishop_solution = solve_bishops_small(bishop_board)
```
Comment (marcosfede) [inline googlecodejam/Fashion Show/fashion_show.py:63]: good catch, I renamed the bishop solution attempting the large dataset and forgot to change it here

File: bit/add_without_operator.py [hunk 2]
```diff
@@ -5,17 +5,24 @@
 Input: 2 3
 Output: 5
 """
+try:
+    raw_input          # Python 2
+except NameError:
+    raw_input = input  # Python 3
```
Comment (marcosfede) [inline bit/add_without_operator.py:11]: Could we modify blocks like this to be python3 only?

File: sort/insertion_sort.py [hunk 3]
```diff
@@ -1,14 +1,20 @@
+try:
+    xrange          # Python 2
+except NameError:
+    xrange = range  # Python 3
```
Comment (marcosfede) [inline sort/insertion_sort.py:4]: We should support python3 only from now on

**Comment by marcosfede** [comment]
Hi, first of all thanks for taking the time to review this!
Looks really good, but I think it would be better to just support python3 syntax and not have only some solutions work around legacy syntax. Could you make those small changes I commented on to merge this?
Thanks!

**Comment by marcosfede** [comment]
Just made a huge update of the problems based on keon's repo so we'll need to merge your fixes with master

**Comment by cclauss** [comment]
Closed in favor of #13.

**Pull request closed without merging**